src/main/java/javax/validation/Validation.java

```diff
@@ -380,7 +387,30 @@ public ValidationProviderResolver getDefaultValidationProviderResolver() {
 		}
 
 		private synchronized void cacheValidationProviders(ClassLoader classLoader, List<ValidationProvider<?>> providers) {
-			providersPerClassloader.put( classLoader, new SoftReference<List<ValidationProvider<?>>>( providers ) );
+			providersPerClassloader.put( classLoader, new SoftReference<>( providers ) );
+		}
+	}
+
+	private static class NewProviderInstance<T extends ValidationProvider<?>> implements PrivilegedAction<T> {
+
+		private final Class<T> clazz;
+
+		public static <T extends ValidationProvider<?>> NewProviderInstance<T> action(Class<T> clazz) {
+			return new NewProviderInstance<>( clazz );
+		}
+
+		private NewProviderInstance(Class<T> clazz) {
```

Comment:
Yes, as Guillaume is saying a proper `ValidationException` is thrown if null has been passed. One could argue wether it should happen a bit earlier, but I don't think there is a strong need to change that.

Regarding `Objects#nonNull`, it's a nice idea, but `Validation` is the only class in the API with an actual implementation (apart from exceptions), so there would not be much cases where it could happen (the one at bay may even be the only one as the other methods are parameterless).

I don't think the spec mandates anything to impls if null is passed for mandatory params, the RI will raise a meaningful exception via its `Contracts` class in this case (it's from a time where there was no `Objects` yet). Long story short, I don't think anything needs to change here.